### PR TITLE
Add `try`-versions of prefix iterators

### PR DESCRIPTION
**Pull request by ark0f**
Can be very useful in migrations to avoid mistakes

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2998668

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3003085

File: frame/support/src/storage/mod.rs [hunk 1]
```diff
@@ -868,6 +868,26 @@ impl<T, OnRemoval1> PrefixIterator<T, OnRemoval1> {
 	}
 }
 
+/// Error returned by `TryPrefixIterator`.
```
Comment (kianenigma) [inline frame/support/src/storage/mod.rs:871]:
```suggestion
/// Error returned by [`TryPrefixIterator`].
```

File: frame/support/src/storage/mod.rs [hunk 2]
```diff
@@ -924,55 +944,116 @@ impl<T, OnRemoval> PrefixIterator<T, OnRemoval> {
...
+		}
+	}
+}
+
+/// Iterate or drain over a prefix and decode raw_key and raw_value into `T`.
+///
+/// If any decoding fails it skips it and continues to the next key.
+///
+/// If draining, then the hook `OnRemoval::on_removal` is called after each removal.
+pub struct PrefixIterator<T, OnRemoval = ()> {
```
[the remark this answers is not in the document]
Comment (ark0f) [reply frame/support/src/storage/mod.rs:989]: Their iterator implementations have different items

**Pull request closed without merging**